doc/components.md

```diff
@@ -1,12 +1,43 @@
 # Components
 
+![Component Diagram](https://i.imgur.com/1mTqcqI.png)
+This diagram might look intimidating, but don't worry, it is not as complex as it looks.
+
+## Component
+All components inherit from the abstract class `Component`.
+
+If you want to skip reading about abstract classes you can jump directly to [PositionComponent](./components.md#PositionComponent).
+
+Every `Component` has a few methods that you can optionally implement, that are used by the `BaseGame` class. If you are not using the base game, you can alternatively use these methods on your own game loop.
+
+The `resize` method is called whenever the screen is resized, and in the beginning once when the component is added via the `add` method. You need to apply here any changes to the x, y, width and height of your component, or any other changes, due to the screen resizing. You can start these variables here, as the sprite won't be rendered until everything is set.
```

Comment:
> as the sprite won't be rendered until everything is set.
this is super outdated as it's referring to a feature of SpriteComponent. it can be very misleading here. let's just remove this sentence altogether.

Reply:
Yeah, it is not even something I have written, I just moved the section. But I'll remove/fix the sentence!